packages/tangle/timemanager.go

```diff
@@ -103,9 +127,8 @@ func (t *TimeManager) Time() time.Time {
 // Synced returns whether the node is in sync based on the difference between TangleTime and current wall time which can
 // be configured via SyncTimeWindow.
 func (t *TimeManager) Synced() bool {
-	t.lastConfirmedMutex.RLock()
-	defer t.lastConfirmedMutex.RUnlock()
-
+	t.lastSyncedMutex.RLock()
+	defer t.lastSyncedMutex.RUnlock()
 	return t.lastSynced
```

Comment:
Alternatively, we could make `lastSynced` an atomic. Or could this cause any issues that when `Synced` is called at the same time of `updateSyncedState`, it wouldn't wait for the update?

Reply:
You are right that an atomic would work too but it seems that handlers of the `SyncChanged` event are supposed to be executed with the assurance that the given updated sync flag is valid during their execution. Of course, examining the event handlers for `SyncChanged` reveals that for them it wouldn't really matter since this flag can change shortly after anyway. But lets keep the lock for now to keep that property.